code/controllers/subsystem/ships.dm

```diff
@@ -116,17 +116,21 @@ var/global/list/ftl_weapons_consoles = list()
 			return
 		if(S.planet != SSstarmap.current_planet)
 			return
-		if(world.time > S.next_attack && S.fire_rate)
-			S.next_attack = world.time + S.fire_rate
-			attack_player(S,pick(get_attacks(S)))
+		for(var/datum/component/weapon/W in S.components)
+			if(world.time > W.next_attack && W.fire_rate)
+				W.next_attack = world.time + W.fire_rate + rand(1,50)
+				attack_player(S, W.attack_data)
 	if(S.attacking_target)
 		if(S.attacking_target.planet != S.planet)
 			return
-		if(world.time > S.next_attack && S.fire_rate)
-			S.next_attack = world.time + S.fire_rate
-			ship_attack(S.attacking_target,S)
+		for(var/datum/component/weapon/W in S.components)
+			if(world.time > W.next_attack && W.fire_rate)
+				W.next_attack = world.time + W.fire_rate + rand(1,50)
+				ship_attack(S.attacking_target,S)
+
+
+/datum/subsystem/ship/proc/ship_attack(var/datum/starship/S,var/datum/ship_attack/attack_data,var/datum/starship/attacker)
```

Comment:
You'll need to refactor ship_attack to use the selected weapon instead of picking a random attack.